test/rules/no-shadowed-variable/test.ts.lint

```diff
@@ -198,8 +211,10 @@ function constructorsWithArgsParamConst(
     derivedCtor: new (...args: any[]) => any,
     baseCtors: Array<new (...args: any[]) => any>,
     args: any[]): void {
-    const args = [];
-          ~~~~       [Shadowed variable: 'args']
+    {
```

Comment:
without the block there would be a compile error because the block scoped variable `args` cannot merge with the function scoped parameter `args`.
The block is needed here because the rule only handles shadowed names only in different scopes